### PR TITLE
Do not fail build if the symbol upload task fails on Android

### PR DESCRIPTION
**Pull request by munkiki7**
Added try/catch block to the gradle symbol upload task.
If it fails, the exception is appended to the respective log file.
When Unity build completes, the post build script will check whether the log file contains the error marker and will write it to the Unity logs.
After that the marker is removed from the log file leaving only the original file content.

### CODE REVIEW

File: src/Sentry.Unity.Editor/Android/PostBuildCheck.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            if (HasError(mappingUploadLogPath, out var mappingError, out var mappingLog))
+            {
+                _logger.LogWarning($"Mapping upload task error: {mappingError}");
+                _logger.LogWarning("Mapping upload log file content:");
+                LogFileContent(mappingLog);
+                File.WriteAllText(mappingUploadLogPath, mappingLog); // Clean up the log file
+            }
+        }
+
+        private bool HasError(string filePath, out string error, out string fileContent)
```
Comment (bitsandfoxes) [inline src/Sentry.Unity.Editor/Android/PostBuildCheck.cs:82]: This method does a lot. Can't we just check for the error marker and dump the whole logfile without any modification?

**Review by bitsandfoxes — CHANGES_REQUESTED** [review]

File: src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs [hunk 2]
```diff
@@ -62,7 +68,15 @@ private string _symbolUploadTaskFormat
...
                 }
+
                 stringBuilder.AppendLine("        }");
+                if (!_isExporting)
+                {
+                    stringBuilder.AppendLine("        } catch (exception) {");
+                    stringBuilder.AppendLine("            def file = new File(logFilePath)");
+                    stringBuilder.AppendLine("            file.append('===ERROR===' + exception)");
+                    stringBuilder.AppendLine("        }");
+                }
```
[the remark this answers is not in the document]
Comment (munkiki7) [reply src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs:79]: Yes

File: src/Sentry.Unity.Editor/Android/PostBuildCheck.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+using UnityEngine;
+
+namespace Sentry.Unity.Editor.Android
+{
+    public class PostBuildCheck : IPostprocessBuildWithReport
+    {
+        public int callbackOrder { get; } = 117;
```
[the remark this answers is not in the document]
Comment (munkiki7) [reply src/Sentry.Unity.Editor/Android/PostBuildCheck.cs:13]: Just a random number :)

File: src/Sentry.Unity.Editor/Android/PostBuildCheck.cs (hunk 1 above)
Comment (munkiki7) [reply src/Sentry.Unity.Editor/Android/PostBuildCheck.cs:82]: I wanted the error message to be printed separately before the whole log is dumped for better user experience.

**Review by bitsandfoxes — APPROVED** [review]

**Pull request merged**